bin/git-mr

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if test "${1-}" = "clean"; then
+  git for-each-ref refs/heads/mr/* --format='%(refname)' | while read ref; do
+    git branch -D ${ref#refs/heads/}
+  done
+  exit 0
+elif [[ ${1-} =~ ^(https?://[^/]+/(.+))/merge_requests/([0-9]+).*$ ]]; then
+  remote=${BASH_REMATCH[1]}.git
+  id=${BASH_REMATCH[3]}
+  branch=mr/$id
+else
+  test -z "${1-}" && echo "mr number required." 1>&2 && exit 1
+  id=$1
+  remote=${2:-origin}
+  branch=mr/$id
+fi
+git fetch -fu $remote refs/merge-requests/$id/head:$branch && \
+git checkout $branch && \
+git config --local --replace branch.$branch.merge refs/mr/$id/head && \
```

Comment:
Well, after running `git fetch -fu $remote refs/merge-requests/1/head:mr/1`, the output of `git config --local branch.mr/1.merge` is  `refs/mr/1/head` by default.

Here is the pitfall.
When you run `git pull origin`(without specific `refspec`), git will use the `branch.$your_current_branch.merge` as the default `refspec`.
However, `refs/mr/1/head` doesn't exist in the remote repo, so `git pull origin` will fail.
To fix this behavior, we need to replace `branch.$branch.merge` to its real name in the remote repo.